client/lib/importer/actions.js

```diff
@@ -149,27 +150,24 @@ export function startImporting( importerStatus ) {
 export function startUpload( importerStatus, file ) {
 	let { importerId, site: { ID: siteId } } = importerStatus;
 
-	wpcom.uploadExportFile( siteId, {
-		importStatus: toApi( importerStatus ),
-		file,
-
-		onload: ( error, data ) => {
-			if ( ! error ) {
-				return finishUpload( importerId, data );
-			}
-
-			failUpload( importerId, error.message );
-		},
-
-		onprogress: event => {
-			setUploadProgress( importerId, {
-				uploadLoaded: event.loaded,
-				uploadTotal: event.total
-			} );
-		},
-
-		onabort: () => cancelImport( siteId, importerId )
-	} );
+	wpcom
+		.uploadExportFile( siteId, {
+			importStatus: toApi( importerStatus ),
+			file,
+
+			onprogress: event => {
+				setUploadProgress( importerId, {
+					uploadLoaded: event.loaded,
+					uploadTotal: event.total
+				} );
+			},
+
+			onabort: () => cancelImport( siteId, importerId )
+		} )
+		.then( data => Object.assign( data, { siteId } ) )
+		.then( fromApi )
+		.then( partial( finishUpload, importerId ) )
+		.catch( partial( failUpload, importerId ) );
```

Comment:
I'm a big fan of using [point-free style](https://en.wikipedia.org/wiki/Tacit_programming) because it helps me see what is happening to the data without getting lost in the variable names and assignments